### PR TITLE
Todo from code etc

### LINKED ISSUE #10: Idea
Would be cool to have a mode where you can get all the todos in a code file e.g.
```
// Some file.ext
...code
// TODO: Fix this crappy code
...code
// TODO: Fix this typo
...code
```
Run
```
todo  file.ext--code
```
or something and get
```
TODO
---------
[1] Fix this crappy code
[2] Fix this typo
```

### LINKED ISSUE #13: Idea: Look for .todo.txt in current path
To support different todo list for different things/projects it would be nice if `todo` by default would look for a local `.todo.txt` in the current directory. This could then work well with #10 where you add all todos for the whole project.

### PR DESCRIPTION
**Pull request by bombsimon**
* Introduce flags
* Support to fetch all `TODO` from current directory recursively
  *  This requires the line to start with 0 or more spaces followed by`#` or `//` followed by `TODO:` (including the colon). The task will be all after `TODO:` and end with `[file]:[line_number]`
* Always reference to lists in `.todo` (no `txt` extension and always hidden)
* Support to auto read `.todo` in the current directory
  * Can be skipped with `--no-local` (shorter than e.g. `--task-file /long/path/to/where/i/usually/store/it`
* Remove `readlink` since symlinks aren't used anymore

Closes #10, Closes #13

### CODE REVIEW

File: todo_from_file.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+# Find all `TODO`-markings in any file.
+grep -rn --exclude-dir={.git,node_modules} "^ *\(#\|//\).*TODO:" . | \
+    sed -E 's/^([0-9]+).*TODO[: ]*/\1:TODO:/' | \
+    awk -F':' '{printf "%s - %s:%s\n", $4, $1, $2}' > .todo
```
Comment (bombsimon) [inline todo_from_file.sh:6]: Maybe backup any existing `.todo` (or show a warning)

File: todo_from_file.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
```
Comment (fippli) [inline todo_from_file.sh:1]: Should we add a flag like `--code`, or something better if you have any idea, for this command? Couldn't find it.
So if I execute `todo --code`, it generates the todo recursively?

Comment (fippli) [reply todo_from_file.sh:1]: Never mind! Missed the `--this-dir`. All is good!

**Review by fippli — APPROVED** [review]
Great! Good job :)

**Pull request merged**